assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -0,0 +1,369 @@
+ */
+function generateMetricValues( validMetrics, count ) {
+	const metrics = [];
+
+	for ( let i = 0; i < count; i++ ) {
+		const values = [];
+
+		validMetrics.forEach( ( validMetric ) => {
+			switch ( getMetricType( validMetric ) ) {
+				case 'INTEGER':
+					values.push(
+						faker.datatype.number( { min: 0, max: 100 } ).toString()
+					);
+					break;
+				case 'PERCENT':
+					values.push(
+						faker.datatype.float( { min: 0, max: 100 } ).toString()
+					);
+					break;
+				case 'TIME':
+					values.push(
+						faker.datatype
+							.number( { min: 0, max: 3600 } )
+							.toString()
+					); // 1 hour max.
+					break;
+				case 'CURRENCY':
+					values.push(
+						faker.datatype
+							.float( { min: 0, max: 10000 } )
+							.toString()
+					); // $10k max.
+					break;
+			}
+		} );
+
+		metrics.push( { values } );
+	}
+
+	return metrics;
+}
+
+/**
+ * Sorts report rows and returns it.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>}        rows    Array of rows to sort.
+ * @param {Array.<Object>}        metrics Array of report metrics.
+ * @param {Object|Array.<Object>} orderby Sorting options.
+ * @return {Array.<Object>} Sorted rows.
+ */
+function sortRows( rows, metrics, orderby ) {
+	let sorted = rows;
+
+	const orders = castArray( orderby );
+	for ( const order of orders ) {
+		const direction = order?.sortOrder === 'DESCENDING' ? -1 : 1;
+		const index = metrics.findIndex(
+			( metric ) => getMetricKey( metric ) === order?.fieldName
+		);
+		if ( index < 0 ) {
+			continue;
+		}
+
+		sorted = sorted.sort( ( a, b ) => {
+			let valA = parseFloat( a.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valA ) ) {
+				valA = 0;
+			}
+
+			let valB = parseFloat( b.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valB ) ) {
+				valB = 0;
+			}
+
+			return ( valA - valB ) * direction;
+		} );
+	}
+
+	return sorted;
+}
+
+/**
+ * Generates mock data for Analytics 4 reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAnalytics4MockResponse( args ) {
+	invariant(
+		isPlainObject( args ),
+		'report options are required to generate a mock response.'
+	);
+	invariant(
+		isValidDateString( args.startDate ),
+		'a valid startDate is required.'
+	);
+	invariant(
+		isValidDateString( args.endDate ),
+		'a valid endDate is required.'
+	);
+
+	const originalSeedValue = faker.seedValue;
+	const argsHash = parseInt(
+		md5( args.url || 'http://example.com' ).substring( 0, 8 ),
+		16
+	);
+
+	// We set seed for every data mock to make sure that the same arguments get the same report data.
+	// It means that everyone will have the same report data and will see the same widgets in the storybook.
+	// This approach gives us additional flexibility to control randomness on a per widget basis.
+	if ( ! Number.isNaN( argsHash ) ) {
+		faker.seed( argsHash );
+	}
+
+	const data = {
```

Comment:
Hi @asvinb, unfortunately a bit of detail was missed in the IB. We need to ensure the response shape matches the new [response schema](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/RunReportResponse). Therefore a bit more work is needed here.

As a concrete reference, using the [API Explorer](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/properties/runReport) I was able to generate an example report:

<details>
<summary>Example report</summary>

```json
{
  "dimensionHeaders": [
    {
      "name": "date"
    }
  ],
  "metricHeaders": [
    {
      "name": "sessions",
      "type": "TYPE_INTEGER"
    },
    {
      "name": "bounceRate",
      "type": "TYPE_FLOAT"
    }
  ],
  "rows": [
    {
      "dimensionValues": [
        {
          "value": "20230101"
        }
      ],
      "metricValues": [
        {
          "value": "55"
        },
        {
          "value": "0.61818181818181817"
        }
      ]
    },
    {
      "dimensionValues": [
        {
          "value": "20230102"
        }
      ],
      "metricValues": [
        {
          "value": "7"
        },
        {
          "value": "0.7142857142857143"
        }
      ]
    }
  ],
  "totals": [
    {
      "dimensionValues": [
        {
          "value": "RESERVED_TOTAL"
        }
      ],
      "metricValues": [
        {
          "value": "62"
        },
        {
          "value": "0.62903225806451613"
        }
      ]
    }
  ],
  "maximums": [
    {
      "dimensionValues": [
        {
          "value": "RESERVED_MAX"
        }
      ],
      "metricValues": [
        {
          "value": "55"
        },
        {
          "value": "0.7142857142857143"
        }
      ]
    }
  ],
  "minimums": [
    {
      "dimensionValues": [
        {
          "value": "RESERVED_MIN"
        }
      ],
      "metricValues": [
        {
          "value": "7"
        },
        {
          "value": "0.61818181818181817"
        }
      ]
    }
  ],
  "rowCount": 2,
  "metadata": {
    "currencyCode": "USD",
    "timeZone": "America/Los_Angeles"
  },
  "kind": "analyticsData#runReport"
}
```

</details>

<details>
<summary>Query for the above report</summary>

```json
{
  "dimensions": [
    {
      "name": "date"
    }
  ],
  "metrics": [
    {
      "name": "sessions"
    },
    {
      "name": "bounceRate"
    }
  ],
  "metricAggregations": [
    "TOTAL",
    "MINIMUM",
    "MAXIMUM"
  ],
  "dateRanges": [
    {
      "startDate": "2023-01-01",
      "endDate": "2023-01-03"
    }
  ]
}
```

</details>

Reply:
Thanks @techanvil . I knew something was kinda missing 😅  I've updated the PR, can you take another look?